### ISSUE TITLE
Sort out process module test failures on macos 10.15

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#764]
I recently disabled the test_process test since it was pretty much always failing on macos 10.15. I'm a little surprised we got through that PR in the first place...

Fix it!

**Comment by plajjan** [comment]
We've stopped testing on macos 10.15, so these failures are no longer blocking anything but perhaps it would be good to understand why the fail. Since implementing the process module I've learnt that I've misunderstood how libuv is supposed to be used. We're not allowed to call any uv functions from other threads, which I think we are, so that could be a first thing to check...

**Comment by plajjan** [comment]
The libuv misimplementation is now fixed through #783 so we pin IO actors to a particular worker thread and thus a particular uv loop!

We no longer have 10.15 in CI and I'm not running it on any computer, so I'm not quite sure how to reproduce.

**Comment by plajjan** [comment]
Finally I've been able to run the docker-osx 10.15 container long enough to reproduce this problem. Not sure why but the container / qemu keeps shutting down all the time due to receiving signal 2, but who is sending signal 2 to it and why? Anyway, the errors happens a lot less frequently on my laptop but with 10000 test runs I was able to get timeout failures 11 times when running with a version (b4f704db11c5f82d527f2b52534ea3c110a0c9de) just prior to the combined libuv WTS threads. After switching to latest main with the combined libuv WTS threads, all 10000 tests invocations go through successfully. Thus I believe this bug was caused by our incorrect use of libuv and since that's corrected, I'll close this issue.
